src/buffer/out/textBuffer.cpp

```diff
@@ -1919,20 +1923,32 @@ HRESULT TextBuffer::Reflow(TextBuffer& oldBuffer,
     // place the new cursor back on the equivalent character in
     // the new buffer.
     const COORD cOldCursorPos = oldCursor.GetPosition();
-    COORD cOldLastChar = lastCharacterViewport.has_value() ?
-                             oldBuffer.GetLastNonSpaceCharacter(lastCharacterViewport.value()) :
-                             oldBuffer.GetLastNonSpaceCharacter();
+    const COORD cOldLastChar = lastCharacterViewport.has_value() ?
+                                   oldBuffer.GetLastNonSpaceCharacter(lastCharacterViewport.value()) :
+                                   oldBuffer.GetLastNonSpaceCharacter();
 
-    short const cOldRowsTotal = cOldLastChar.Y + 1;
-    short const cOldColsTotal = oldBuffer.GetSize().Width();
+    const short cOldRowsTotal = cOldLastChar.Y + 1;
+    const short cOldColsTotal = oldBuffer.GetSize().Width();
 
     COORD cNewCursorPos = { 0 };
     bool fFoundCursorPos = false;
-
+    bool foundOldRow = false;
     HRESULT hr = S_OK;
     // Loop through all the rows of the old buffer and reprint them into the new buffer
     for (short iOldRow = 0; iOldRow < cOldRowsTotal; iOldRow++)
     {
+        // If we found the old row that the caller was interested in, set the
+        // out value of that parameter to the cursor's current Y position (the
+        // new location of that row in the buffer).
+        if (oldViewportTop && !foundOldRow)
+        {
+            if (iOldRow >= *oldViewportTop)
+            {
+                *oldViewportTop = newCursor.GetPosition().Y;
+                foundOldRow = true;

```

Comment:
```suggestion
                break;
```

Reply:
And you could probably just refactor this section a bit to just get rid of `foundOldRow` entirely.

Reply:
wait that won't work at all, there's so much other work that gets done in this loop...

Reply:
Idk why I thought that the whole for-loop was just the if-statement. My bad!